regenerate-initrd-posttrans

```diff
@@ -93,14 +189,37 @@ else
 			else
 				work_done=yes
 			fi
-		elif ! "$DRACUT" -f --kver "$kver"; then
-			err=$?
 		else
-			work_done=yes
+			case "$initrd_generator" in
+				"dracut")
+					if ! "$DRACUT" -f --kver "$kver"; then
+						err=$?
+					else
+						work_done=yes
+					fi
+					;;
+				"mkosi-initrd")
+					if ! "$MKOSI_INITRD" --kernel-version "$kver" -O "$MKOSI_INITRD_STAGING_DIR" -o "initrd-$kver"; then
+						err=$?
+					elif ! chmod 600 "$MKOSI_INITRD_STAGING_DIR/initrd-$kver"; then
+						err=$?
+					elif ! cp --reflink=auto "$MKOSI_INITRD_STAGING_DIR/initrd-$kver" "/boot/initrd-$kver"; then
```

Comment:
Is `--reflink=auto` portable? It's the default anyway for GNU cp, no?

Reply:
dracut explicitly uses `--reflink=auto` everywhere since forever, I didn't check if it's the default now. I can remove it if you want, but it doesn't hurt IMO.

Reply:
Ok then. Just verified that `busybox cp` at least doesn't bail out on this option.